doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,290 @@
+the same effect. However, the latter is more space-efficient. Quote symbol
+quoting is handled by regular SQL quoting. Maximum supported dictionary length
+is 32506 bytes (``zlib`` limitation).
+
+To achieve better compression ration on relatively small individual data items,
+it is possible to pre-define compression dictionary, which is a set of strings
+for each compressed column.
+
+The compression dictionary is stored in a new system |InnoDB| table.
+As this table is of the data dictionary kind, concurrent reads are
+allowed, but writes are serialized, and reads are blocked by writes. Table read
+through old read views are unsupported, similarly to |InnoDB| internal DDL
+transactions.
+
+Example
+-------
+
+In order to use the compression dictionary you'll need first to create it. This
+can be done by running:
+
+.. code-block:: mysql
+
+   mysql> SET @dictionary_data = 'one' 'two' 'three' 'four';
+   Query OK, 0 rows affected (0.00 sec)
+
+   mysql> CREATE COMPRESSION_DICTIONARY numbers (@dictionary_data);
+   Query OK, 0 rows affected (0.00 sec)
+
+To create a table that has both compression and compressed dictionary support
+you should run:
+
+.. code-block:: mysql
+
+   mysql> CREATE TABLE t1(
+           id INT,
+           a BLOB COLUMN_FORMAT COMPRESSED,
+           b BLOB COLUMN_FORMAT COMPRESSED WITH COMPRESSION_DICTIONARY numbers
+         ) ENGINE=InnoDB;
+
+Following example shows how to insert a sample of JSON data into the table:
+
+.. code-block:: mysql
+
+  SET @json_value =
+   '[\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' },\n'
+   ' {\n'
+   ' "one" = 0,\n'
+   ' "two" = 0,\n'
+   ' "three" = 0,\n'
+   ' "four" = 0\n'
+   ' }\n'
+   ']\n'
+  ;
+
+.. code-block:: mysql
+
+  mysql> INSERT INTO t1 VALUES(0, @json_value, @json_value);
+  Query OK, 1 row affected (0.01 sec)
+
+
+INFORMATION_SCHEMA Tables
+=========================
+
+This feature implemented two new ``INFORMATION_SCHEMA`` tables.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT
+
+  :column BIGINT(21)_UNSIGNED id: dictionary ID
+  :column VARCHAR(64) name: dictionary name
+  :column BLOB zip_dict: compressed dictionary
+
+This table stores the compression dictionary itself. ``SUPER`` privilege is
+required to query it.
+
+.. table:: INFORMATION_SCHEMA.XTRADB_ZIP_DICT_COLS
+
+  :column BIGINT(21)_UNSIGNED table_id: table ID from ``INFORMATION_SCHEMA.INNODB_SYS_TABLES``
+  :column BIGINT(21)_UNSIGNED column_pos: column position (starts from ``0`` as in ``INFORMATION_SCHEMA.INNODB_SYS_COLUMNS``)
+  :column BIGINT(21)_UNSIGNED dict_id: dictionary ID
+
+This table stores the mapping between the compression dictionary and columns
```

Comment:
This table provides a view over the internal table that stores the mapping between the compression dictionaries and the columns using them.